### PR TITLE
test: update test naming conventions

### PR DESCRIPTION
**Pull request by avasisht23**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, see guidleines below)
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:fix`)
- [x] Is the base branch you're merging into `development` and not `main`?

---

### Focus of this PR:
This PR focuses on updating code to use the `provider` variable instead of the `signer` variable for signing messages, encoding batch transaction data, and other related operations.

### Detailed summary:
- Updated code to use `provider` instead of `signer` for signing messages and encoding batch transaction data.
- Updated code to use `API_KEY` instead of `process.env.API_KEY!` in `constants.ts` files.
- Updated code to use `chain` variable instead of `polygonMumbai` in various files.
- Updated code to use `getDefaultLightAccountFactoryAddress()` function instead of hardcoding the factory address in `account.test.ts` file.
- Updated code to use the correct Alchemy SDK network in `provider.test.ts` file.

> The following files were skipped due to too many changes: `packages/alchemy/src/__tests__/provider.test.ts`, `packages/core/e2e-tests/simple-account.test.ts`, `packages/alchemy/src/__tests__/__snapshots__/provider.test.ts.snap`, `packages/alchemy/e2e-tests/simple-account.test.ts`, `packages/accounts/src/light-account/e2e-tests/light-account.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by avasisht23** [comment]
Current dependencies on/for this PR:
* development
  * **PR #242** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/242?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #206** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/206?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/242?utm_source=stack-comment).

File: packages/core/e2e-tests/constants.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
-export const MUMBAI_RPC_URL = process.env.MUMBAI_RPC_URL!;
```
Comment (avasisht23) [inline packages/core/e2e-tests/constants.ts:1]: not sure when we did this - but putting it back to api key only. we should hard code the rpc url in code.

File: packages/alchemy/e2e-tests/simple-account.test.ts [hunk 2]
```diff
@@ -173,26 +183,7 @@ describe("Simple Account Tests", () => {
...
-          "contractAddress": "0xdcf5d3e08c5007dececdb34808c49331bd82a247",
-          "tokenBalance": "0x00000000000000000000000000000000000000000000000000000000000f423f",
-        },
-        {
-          "contractAddress": "0xfff9976782d46cc05630d1f6ebab18b2324d6b14",
-          "tokenBalance": "0x0000000000000000000000000000000000000000000000000000000000000000",
-        },
-      ]
-    `);
+    expect(balances.tokenBalances.length).toMatchInlineSnapshot(`4`);
```
Comment (avasisht23) [inline packages/alchemy/e2e-tests/simple-account.test.ts:186]: simplifying the check.

File: packages/accounts/src/kernel-zerodev/e2e-tests/constants.ts [hunk 3]
```diff
@@ -1,2 +1,2 @@
-export const MUMBAI_RPC_URL = process.env.MUMBAI_RPC_URL;
```
Comment (avasisht23) [inline packages/accounts/src/kernel-zerodev/e2e-tests/constants.ts:1]: not sure when we did this - but putting it back to api key only. we should hard code the rpc url in code.

**Review by avasisht23 — COMMENTED** [review]
At the moment, we were coupling "signer" and "provider" in our tests. The provider is responsible for much more context than signing.

This is to disambiguate naming conventions of "signer" from "provider" as they hope to serve as an example for developers looking to integrate.

**Review by denniswon — APPROVED** [review]
can you share the .env file that you use for testing? Can we upload it to 1Password? wdyt? @moldy530  @avasisht23

**Pull request merged**